R/getDegreeDays.R

```diff
@@ -0,0 +1,206 @@
+                          tLim = list("HDD" = seq(9, 19), "CDD" = seq(15, 25)),
+                          std  = c("tLim" = 2, "tAmb" = 2),
+                          ssp  = c("historical", "SSP2"),
+                          outDir = "output") {
+  # CHECKS ---------------------------------------------------------------------
+
+  # check for mapping file
+  if (is.null(mappingFile)) {
+    stop("No mapping file was provided.")
+  }
+
+  # absolute path
+  mappingFile <- getSystemFile("extdata", "mappings", mappingFile, package = "climbed")
+
+  if (!file.exists(mappingFile)) {
+    stop("Provided mapping file does not exist in /extdata/mappings.")
+  }
+
+
+
+  # PARAMETERS -----------------------------------------------------------------
+
+  # process arguments
+  args <- processArgs(tLim = tLim, std = std, ssp = ssp)
+
+  # se standardized arguments
+  tLim <- args$tLim
+  std  <- args$std
+  ssp  <- args$ssp
+
+
+  # range of pre-calculated HDD/CDD-values, e.g. [173, 348] K, converted to [C]
+  tLow <- 173 - 273.15
+  tUp  <- 348 - 273.15
+
+
+  # track submitted jobs
+  allJobs <- list()
+
+
+
+  # READ-IN DATA ---------------------------------------------------------------
+
+  # file mapping
+  fileMapping <- read.csv2(mappingFile)
+
+
+  # external BAIT weights and parameters
+  wBAIT <- read.csv2(getSystemFile("extdata", "mappings", "BAITweights.csv",
+                                   package = "climbed"),
+                     stringsAsFactors = FALSE)
+
+
+  # population files
+  popMapping <- read.csv2(getSystemFile("extdata", "mappings", "populationMapping.csv",
+                                        package = "climbed"),
+                          stringsAsFactors = FALSE)
+
+
+  # scenario matrix
+  scenMatrix <- read.csv2(getSystemFile("extdata", "mappings", "scenarioMatrix.csv",
+                                        package = "climbed"),
+                          stringsAsFactors = FALSE)
+
+
+
+  # CHECKS ---------------------------------------------------------------------
+
+  # check if mapping file contains correct columns
+  mappingCols <- c("gcm", "rcp", "start", "end", "tas", "rsds", "sfc", "huss")
+  if (!any(mappingCols %in% colnames(fileMapping))) {
+    stop("Please provide file mapping with correct columns.\n Missing columns: ")
+  }
+
+  # create output directory if it doesn't exist
+  dir.create(outDir, showWarnings = FALSE, recursive = TRUE)
+
+  # create output logs directory if it doesn't exist
+  dir.create(file.path(outDir, "logs"), showWarnings = FALSE, recursive = TRUE)
+
+
+
+  # PROCESS DATA ---------------------------------------------------------------
+
+  # --- Prepare Mappings
+
+  # BAIT weights
+  wBAIT <- setNames(as.list(wBAIT$value), wBAIT$variable)
+
+  # population data
+  popMapping <- setNames(as.list(popMapping$file), popMapping$scenario)
+
+  # scenario matrix
+  scenMatrix <- setNames(lapply(strsplit(scenMatrix$rcp, ","), trimws), scenMatrix$ssp)
```

Comment:
Can't you use `dplyr::pull` here?

Reply:
Sure, but this would only replace the ´scenMatrix$...´ entries, what benefit would there be? An item of ´scenMatrix$rcp´ looks like "1.9, 2.6, ..." and thus needs to be split accordingly.

Reply:
You might be right. I still think this could improve readibility a bit...
```suggestion
  wBAIT <- wBAIT %>%
    pull("value", "variable") %>%
    as.list()

  # population data
  popMapping <- popMapping  %>%
    pull("file", "scenario") %>%
    as.list()

  # scenario matrix
  scenMatrix <- scenMatrix %>%
    pull("rcp", "ssp") %>%
    strsplit(",") %>%
    lapply(trimws)
```